src/App.js

```diff
@@ -1,17 +1,197 @@
+        words.splice(r,1);
+      }
+      let newState = {...this.state};
+      newState.word = word;
+      newState.hint = hint;
+      newState.score ={value: 100, className: "high-score"};
+      newState.letterStatus = this.generateLetterStatus();
+      this.setState({...this.state,word:newState.word,
+        score:newState.score,
+        letterStatus:newState.letterStatus,
+        hint:newState.hint
+      });
+    }
+    
+    resetLife = ()=> {
+      toShow = false; //hide
+      let newState = {...this.state};
+      newState.score ={value: 100, className: "high-score"};
+      newState.letterStatus = this.generateLetterStatus();
+      this.setState({...this.state,
+        score:newState.score,
+        letterStatus:newState.letterStatus
+      });
+    }
+    
+    generateLetterStatus() {
+      let letterStatus = {};
+      for (let i = 65; i < 91; i++) {
+        letterStatus[String.fromCharCode(i)] = false;
+      }
+      return letterStatus;
+    }
+    
+    selectLetter = letter => {
+      let state =  { ...this.state}
+      let letterStatus = state.letterStatus;
+      //let letterStatus = this.state.letterStatus - > never use it like this
+      letterStatus[letter] = true;
+      //state.word.indexOf(letter) > -1 ? this.updateScore(5): this.updateScore(-20);
+      (this.state.word.includes(letter))?this.updateScore(5):this.updateScore(-20);
+      this.setState({ letterStatus: letterStatus }); //the right-hand side is the updated object! 
+    };
+    
+    //return new object - just one update the status
+    //For each right letter, increase the score by 5
+    //For each wrong letter, decrease the score by 20
+    updateScore = update => {
+      let newScore = { ...this.state.score };
+      newScore.value += update;
+      if (newScore.value >= 80) {
+        toShow = false; //hide
+        newScore.className = "high-score";
+        this.setState({ ...this.state, score: newScore });
+      } else if (newScore.value < 80 && newScore.value > 60) {
+        toShow = false; //hide
+        newScore.className = "mid-score";
+        this.setState({ ...this.state, score: newScore });
+      } else if (newScore.value <= 60) {
+        toShow = true; //show
+        newScore.className = "low-score";
+        this.setState({ ...this.state, score: newScore });
+      }
+    };
+    
+    isFinishGame = () => {
+      for (let letter of this.state.word) {
+        if (!this.state.letterStatus[letter]){
+          return false
+        }
+      }
+      return true;
+    };
+    
+    showMeLetter = () =>{
+      toShow = false;
+      let state =  {...this.state}
+      let letterStatus = state.letterStatus;
+      for (let l of this.state.word) {
+        if (!letterStatus[l]){
+          letterStatus[l]=true
+          this.setState({ ...this.state, letterStatus: letterStatus });
+          return;
+        }
+      }
+    }
+    
+    
+    
+    render() {
+      if (this.state.score.value <= 0) {
```

Comment:
extract each of the `return(...)` to separate components (or at least generate them in a class method), then return them conditionally

the render method should be pretty minimalist; we shouldn't be building much JSX here